java/src/main/java/com/google/crypto/tink/streamingaead/StreamingAeadKeyTemplates.java

```diff
@@ -0,0 +1,128 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////////
+
+package com.google.crypto.tink.streamingaead;
+
+import com.google.crypto.tink.proto.AesCtrHmacStreamingKeyFormat;
+import com.google.crypto.tink.proto.AesCtrHmacStreamingParams;
+import com.google.crypto.tink.proto.AesGcmHkdfStreamingKeyFormat;
+import com.google.crypto.tink.proto.AesGcmHkdfStreamingParams;
+import com.google.crypto.tink.proto.HashType;
+import com.google.crypto.tink.proto.HmacParams;
+import com.google.crypto.tink.proto.KeyTemplate;
+import com.google.crypto.tink.proto.OutputPrefixType;
+
+/**
+ * Pre-generated {@code KeyTemplate} for {@code StreamingAead} keys. One can use these templates to
+ * generate new {@code Keyset} with {@code KeysetHandle}. To generate a new keyset that contains a
+ * single {@code AesGcmHkdfStreamingKey}, one can do:
+ *
+ * <pre>
+ *   Config.register(StreamingAeadConfig.TINK_1_1_0);
+ *   KeysetHandle handle =
+ *       KeysetHandle.generateNew(StreamingAeadKeyTemplates.AES128_GCM_HKDF_STREAMING);
+ *   StreamingAead streamingAead = StreamingAeadFactory.getPrimitive(handle);
+ * </pre>
+ */
+public final class StreamingAeadKeyTemplates {
+
+  /**
+   * A {@code KeyTemplate} that generates new instances of {@code AesGcmHkdfStreamingKey} with the
+   * following parameters:
+   *   - AES key size: 16 bytes
+   *   - AES IV size: 16 bytes
+   *   - HMAC tag size: 16 bytes
+   *   - Ciphertext segment size: 256
```

Comment:
256 is a bit too small. I think we should use 4KB, which is usually the size of a memory page in most operating systems.